### PR TITLE
ci(test-matrix): multi-platform tests

### PR DESCRIPTION
**Pull request by uguy**
Adding some more integration tests using os matrix support

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -7,11 +7,51 @@ on:
...
+    steps:
+      - name: Install Java and Maven
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+
+      - name: Generate project and build it
+        shell: bash
+        run: >-
+            mvn archetype:generate
```
Comment (educhastenier) [inline .github/workflows/build.yml:44]: you could add `--no-transfer-progress` to remove all those download stack that pollutes the log output...

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -7,11 +7,51 @@ on:
...
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+    - id: get-app-version
+      name: Get archetype version
+      run: echo "::set-output name=APP_VERSION::$(mvn -q -Dexec.executable=echo -Dexec.args='${project.version}' --non-recursive exec:exec)"
     - name: Build and test archetype
       run: ./mvnw clean install
```
Comment (educhastenier) [inline .github/workflows/build.yml:27]: From Maven 3.6.1, you can add `--no-transfer-progress` to remove all those download stack that pollutes the log output...

**Comment by uguy** [comment]
This PR comes after some people had problem generating or building a connector project. 
We try to ensure it's working at least on the latest version of Ubuntu, MacOS and Windows.

 The workflow goes like this : 
1 - First
* build archetype
* _"publish it"_ (stored in workflow)

2 - Then, for each test OS
* _"download and install"_ archetype
* generate and build/test project

**Comment by rbioteau** [comment]
Looks good to me

**Pull request merged**